### ISSUE TITLE
Unsupported arity in ActiveRecord 7

### ISSUE THREAD

**Issue by headius** [jruby/jruby#7428]
```
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

wrong number of arguments (given 1, expected 0)
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activerecord-7.0.4/lib/active_record/persistence.rb:1064:in `create_or_update'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activerecord-7.0.4/lib/active_record/callbacks.rb:457:in `block in create_or_update'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activesupport-7.0.4/lib/active_support/callbacks.rb:99:in `run_callbacks'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activesupport-7.0.4/lib/active_support/callbacks.rb:929:in `_run_save_callbacks'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activerecord-7.0.4/lib/active_record/callbacks.rb:457:in `create_or_update'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activerecord-7.0.4/lib/active_record/timestamp.rb:126:in `create_or_update'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activerecord-7.0.4/lib/active_record/persistence.rb:648:in `save!'
/home/headius/work/jruby/lib/ruby/gems/shared/gems/activerecord-7.0.4/lib/active_record/validations.rb:53:in `save!'

```

**Comment by headius** [comment]
This seems to reproduce the error:

```
class A
  def foo(**)
    p 1
  end
end

class B < A
  def foo(**)
    1.times { super }
  end
end
```

Compared with CRuby:
```
[] ~/work/jruby/testapp $ jruby -e 'class A; def foo(**); p 1; end; end; class B < A; def foo(**); 1.times { super }; end; end; B.new.foo(a: 1)'
ArgumentError: wrong number of arguments (given 1, expected 0)
     foo at -e:1
     foo at -e:1
   times at org/jruby/RubyFixnum.java:302
     foo at -e:1
  <main> at -e:1

[] ~/work/jruby/testapp $ rvm ruby-3.1 do ruby -e 'class A; def foo(**); p 1; end; end; class B < A; def foo(**); 1.times { super }; end; end; B.new.foo(a: 1)'
1

**Comment by enebo** [comment]
Fixed in  #7431.
